lambda/matching.ml

```diff
@@ -567,9 +587,12 @@ let rec flatten_pat_line size p k =
   | Tpat_or (p1, p2, _) ->
       flatten_pat_line size p1 (flatten_pat_line size p2 k)
   | Tpat_alias (p, _, _) ->
-      (* Note: if this 'as' pat is here, then this is a
-                           useless binding, solves PR#3780 *)
+      (* Note: we are only called from flatten_matrix,
+         which is itself only ever used in places
+         where variables do not matter (default environments,
+         "provenance", etc.). *)
       flatten_pat_line size p k
+  | Tpat_var _ -> Patterns.omegas size :: k
```

Comment:
Why not use a or-pattern add to the `Tpat_any` in place of this extra clause ?

Reply:
done